python/lsst/pipe/base/mermaid_tools.py

```diff
@@ -0,0 +1,431 @@
+    ds_id : str
+        Unique Mermaid node identifier for this dataset.
+    ds_name : str
+        The dataset type name.
+    connection : BaseConnection
+        The dataset connection object, potentially dimensioned and having a
+        storage class.
+    universe : DimensionUniverse
+        Used to conform and sort the dataset's dimensions if it is dimensioned.
+    file : file-like
+        The output file-like object to write the Mermaid node definition.
+    show_dimensions : bool
+        If True, display the dataset's conformed dimensions.
+    show_storage : bool
+        If True, display the dataset's storage class if available.
+    """
+    # Start with the dataset name in bold.
+    lines = [f"<b>{html.escape(ds_name)}</b>"]
+
+    # If dimensioned and requested, show conformed dimensions.
+    ds_dims = []
+    if show_dimensions and isinstance(connection, connectionTypes.DimensionedConnection):
+        ds_dims = _expand_dimensions(connection.dimensions, universe)
+
+    if ds_dims:
+        dim_str = _format_dimensions(ds_dims)
+        lines.append(f"<i>dimensions:</i>&nbsp;{html.escape(dim_str)}")
+
+    # If storage class is available and requested, display it.
+    if show_storage and getattr(connection, "storageClass", None) is not None:
+        lines.append(f"<i>storage class:</i>&nbsp;{html.escape(str(connection.storageClass))}")
+
+    label = "<br>".join(lines)
+    print(f'{ds_id}["{label}"]', file=file)
+    print(f"class {ds_id} ds;", file=file)
+
+
+def pipeline2mermaid(
+    pipeline: Pipeline | Iterable[TaskDef], file: Any, show_dimensions: bool = True, show_storage: bool = True
+) -> None:
+    """Convert a Pipeline into a Mermaid flowchart diagram.
+
+    This function produces a Mermaid flowchart, representing tasks and their
+    inputs/outputs as dataset nodes. It uses a top-down layout.
+
+    Parameters
+    ----------
+    pipeline : Pipeline or Iterable[TaskDef]
+        The pipeline or collection of tasks to represent.
+    file : str or file-like
+        The output file or file-like object into which the Mermaid code is
+        written.
+    show_dimensions : bool, optional
+        If True, display dimension information for tasks and datasets.
+        Default is True.
+    show_storage : bool, optional
+        If True, display storage class information for datasets. Default is
+        True.
+
+    Raises
+    ------
+    OSError
+        Raised if the output file cannot be opened.
+    ImportError
+        Raised if the task class cannot be imported.
+    """
+    universe = DimensionUniverse()
+
+    # Ensure that pipeline is iterable of task definitions.
+    if isinstance(pipeline, Pipeline):
+        pipeline = pipeline.to_graph()._iter_task_defs()
+
+    # Open file if needed.
+    close = False
+    if not hasattr(file, "write"):
+        file = open(file, "w")
+        close = True
+
+    # Begin the Mermaid code block with top-down layout.
+    print("```mermaid", file=file)
```

Comment:
As above, I think this is markdown syntax, rather than something in the mermaid schema. Is that correct?